rotkehlchen/db/accounting_rules.py

```diff
@@ -336,3 +303,151 @@ def query_rules_and_serialize(
+        related_events: Sequence[HistoryBaseEntry],
+        event: HistoryBaseEntry,
+        pot: 'AccountingPot',
+) -> bool:
+    """Check if an event will be affected by a callback or treatment from a previous event"""
+    for related_event_idx, related_event in enumerate(related_events):
+        counterparty = getattr(related_event, 'counterparty', None)
+        if counterparty == CPT_GAS:  # avoid checking the case of gas in evm events
+            continue
+
+        # query the accounting rule for the related event
+        cursor.execute(
+            'SELECT accounting_treatment FROM accounting_rules WHERE (type=? AND subtype=? AND (counterparty=? OR counterparty=?))',  # noqa: E501
+            (
+                related_event.event_type.serialize(),
+                related_event.event_subtype.serialize(),
+                NO_ACCOUNTING_COUNTERPARTY if counterparty is None else counterparty,
+                NO_ACCOUNTING_COUNTERPARTY,
+            ),
+        )
+
+        if (raw_treatment := cursor.fetchone()) is not None and raw_treatment[0] is not None:
+            accounting_treatment = TxAccountingTreatment.deserialize_from_db(raw_treatment[0])
+            match accounting_treatment:
+                case TxAccountingTreatment.SWAP:
+                    # ensure that the event that we are going to check can exist
+                    # in the related events. Avoid index error
+                    if len(related_events) <= related_event_idx + 1:
+                        continue
+
+                    if (
+                        related_events[related_event_idx + 1].identifier == event.identifier
+                    ):
+                        return True
+                case TxAccountingTreatment.SWAP_WITH_FEE:
+                    if len(related_events) <= related_event_idx + 2:
+                        continue
+
+                    if event.identifier in {  # check both for the fee and the swap event
+                        related_events[related_event_idx + 1].identifier,
+                        related_events[related_event_idx + 2].identifier,
+                    }:
+                        return True
+
+        if isinstance(related_event, EvmEvent) is False:
+            continue  # only evm events have callbacks
+
+        # if there is no accounting treatment check for callbacks
+        if (callback_data := callbacks.get(related_event.get_type_identifier())) is None:
+            continue
+
+        affected_events_num, callback = callback_data
+        if affected_events_num == 0:
+            continue
+
+        # count the number of events that will be consumed
+        consumed_events = callback(
+            pot=pot,
+            event=related_event,  # type: ignore[arg-type] # mypy doesn't recognize that this is a evm event
+            other_events=iter(related_events[related_event_idx + 1:]),  # type: ignore[arg-type]  # mypy doesn't recognize that this is a evm event
+        )
+
```

Comment:
**important**: After going through the logic of this function what are you doing here? Why are you re-iterating the events once you see that `N` events need to be processed. At this point you should just return the number of processed events from this function.

This function should return amount of events to skip. not a boolean.